src/linear_algebra/exterior_algebra.lean

```diff
@@ -0,0 +1,152 @@
+/-
+Copyright (c) 2020 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Zhangir Azerbayev, Adam Topaz, Eric Wieser.
+-/
+
+import algebra.ring_quot
+import .tensor_algebra
+import group_theory.perm.sign
+
+/-!
+# Exterior Algebras
+
+We construct the exterior algebra of a semimodule `M` over a commutative semiring `R`.
+
+## Notation
+
+The exterior algebra of the `R`-semimodule `M` is denoted as `exterior_algebra R M`.
+It is endowed with the structure of an `R`-algebra. Some results only hold at the level of
+generality of an `S`-module `M` and the space `exterior_algebra S N`.
+
+Given a linear morphism `f : M → A` from a semimodule `M` to another `R`-algebra `A`, such that
+`cond : ∀ m : M, f m * f m = 0`, there is a (unique) lift of `f` to an `R`-algebra morphism,
+which is denoted `exterior_algebra.lift R M f cond`.
+
+The canonical linear map `M → exterior_algebra R M` is denoted `exterior_algebra.ι R M`.
+
+## Theorems
+
+The main theorems proved ensure that `exterior_algebra R M` satisfies the universal property
+of the exterior algebra.
+1. `ι_comp_lift` is  fact that the composition of `ι R M` with `lift R M f cond` agrees with `f`.
+2. `lift_unique` ensures the uniqueness of `lift R M f cond` with respect to 1.
+
+## Implementation details
+
+The exterior algebra of `M` is constructed as a quotient of the tensor algebra, as follows.
+1. We define a relation `exterior_algebra.rel R M` on `tensor_algebra R M`.
+  This is the smallest relation which identifies squares of elements of `M` with `0`,
+  and which is compatible with addition and multiplication.
+2. The exterior algebra is the quotient of the tensor algebra by this relation.
+
+-/
+
+variables (R : Type*) [comm_semiring R]
+variables (M : Type*) [add_comm_monoid M] [semimodule R M]
+
+namespace exterior_algebra
+open tensor_algebra
+
+/--
+An inductively defined relation on `tensor_algebra R M` used to define the exterior algebra.
+-/
+inductive rel : tensor_algebra R M → tensor_algebra R M → Prop
+| of (m : M) : rel ((ι R M m) * (ι R M m)) 0
+
+end exterior_algebra
+
+/--
+The exterior algebra of an `R`-semimodule `M`.
+-/
+@[derive [inhabited, semiring, algebra R]]
+def exterior_algebra := ring_quot (exterior_algebra.rel R M)
+
+namespace exterior_algebra
+
+/--
+The canonical quotient map `tensor_algebra R M → exterior_algebra R M`.
+-/
+protected def quot : tensor_algebra R M →ₐ[R] exterior_algebra R M :=
+  ring_quot.mk_alg_hom R _
```

Comment:
Here and below, don't indent the first line after `:=`. (Indenting happens *before* the `:=` when the theorem statement runs over more than one line, but then the definition/proof starts again in the first column.)

Reply:
Removed this definition entirely, it wasn't used and I'm not yet convinced it was useful.